src/transformers/models/blt_wip/modeling_blt.py

```diff
@@ -0,0 +1,1457 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+import logging
+import os
+from typing import Callable, List, Optional, Tuple, Union
+
+from ...cache_utils import Cache
+
+import torch
+import torch.nn
+import torch.nn as nn
+from torch.nn import functional as F
+from torch.nn.attention.flex_attention import BlockMask, create_block_mask, flex_attention
+
+from ...modeling_rope_utils import ROPE_INIT_FUNCTIONS, dynamic_rope_update
+
+from ...modeling_utils import ALL_ATTENTION_FUNCTIONS, PreTrainedModel
+from .configuration_blt import (
+    BLTConfig,
+    PatchingModeEnum,
+)
+
+RMSNorm = nn.RMSNorm
+
+logger = logging.getLogger()
```

Comment:
We use our own custom logging system in transformers to have more granular control

```suggestion
logger = logging.get_logger(__name__)
```